relayer/src/link.rs

```diff
@@ -1356,38 +1352,49 @@ impl RelayPath {
         Ok(())
     }
 
-    /// Pulls out the next operational data that has fulfilled the predefined delay period and can
+    /// Pulls out the next operational that has fulfilled the predefined delay period and can
     /// now be processed. Does not block: if no OD fulfilled the delay period (or none is
     /// scheduled), returns immediately with `None`.
-    fn try_fetch_scheduled_operational_data(&mut self) -> Option<OperationalData> {
+    fn try_fetch_scheduled_operational_data(
+        &mut self,
+    ) -> (Option<OperationalData>, Option<OperationalData>) {
         // The head of the op. data vector contains the oldest entry.
-        if let Some(od_data_src) = self.src_operational_data.first() {
-            if od_data_src.scheduled_time.elapsed() > self.channel.connection_delay {
-                let op_src = self.src_operational_data.remove(0);
-                info!(
-                    "[{}] Found operational data for source with size: {} (delayed by: {:#?})",
-                    self,
-                    op_src.batch.len(),
-                    op_src.scheduled_time.elapsed()
-                );
-                return Some(op_src);
-            }
-        }
-
-        // No operational data was found targeting the source chain; try for the destination chain.
-        if let Some(od_data_dst) = self.dst_operational_data.first() {
-            if od_data_dst.scheduled_time.elapsed() > self.channel.connection_delay {
-                let op_dst = self.dst_operational_data.remove(0);
-                info!(
-                    "[{}] Found operational data for destination with size: {} (delayed by: {:#?})",
-                    self,
-                    op_dst.batch.len(),
-                    op_dst.scheduled_time.elapsed()
-                );
-                return Some(op_dst);
-            }
-        }
-        None
+        (
+            match self.src_operational_data.first() {
```

Comment:
Can we pull each of the tuple member into their own `let` binding and rewrite the matches as a `flat_map` to improve readability?